modules/nextflow/src/main/groovy/nextflow/file/PathVisitor.groovy

```diff
@@ -103,6 +103,27 @@ class PathVisitor {
             target.bind(STOP)
     }
 
+    private void applyPathPattern0(Path filePattern) {
+        if( isQuery(filePattern) )
+            applyQueryablePath0(filePattern as QueryablePath)
+        else
+            applyGlobPattern0(filePattern)
+    }
+
+    private static boolean isQuery(Path filePattern) {
+        log.debug("Checking if query: $filePattern.class ")
+        return filePattern instanceof QueryablePath && (filePattern as QueryablePath).hasQuery()
+    }
+
+    private boolean applyQueryablePath0(QueryablePath path) {
```

Comment:
We don't need to add query support to `fromPath` because it will already be supported through `queryLineage`. Besides I would like to get away from using URI query params everywhere.

The `queryLineage` factory should return a channel of metadata objects, so you can chain it with a `map` operator to extract the actual files

Reply:
> Maybe the best is returning the lid, letting the user do what they want according to the query.

This is exactly it. The `lineage` function and `queryLineage` factory should return just the metadata descriptions. Users can compose these with other operators to do things like get the actual files.

Meanwhile, `fromPath` should only be used to get individual files.